doc/prelims.rst

```diff
@@ -22,39 +25,75 @@ Running a **Linux** distribution, Python must be compiled. Download and compile
 .. code-block:: shell
    
    wget https://www.python.org/ftp/python/3.6.1/Python-3.6.1.tar.xz
-   sudo tar xzf Python-3.6.1.tar.xz
+   tar xvf Python-3.6.1.tar.xz   # sudo might be required
    cd Python-3.6.1
-   ./configure       # sudo might be required
-   make altinstall   # sudo might be required
+   ./configure --with-ensurepip=install   # sudo might be required
+   make
+   make test
+   make install   # sudo might be required
 
-Some **newer Linux** distributions may be able to install Python 3.6+ using your package manager.
+Some **newer Linux distributions** may be able to install Python 3.6+ using a package manager:
 
 .. code-block:: shell
 
    apt-get install python3.6   # sudo might be required
 
+Running **macOS** install/update Python 3.0+ (here using the reccomended `Homebrew <https://brew.sh/>`__ package manager):
 
-2) Install virtualenvwrapper
-----------------------------
+.. code-block:: shell
+
+   brew install python3
+   brew upgrade python3 
+
+A .profile file is a shell script that Bash runs whenever it is started or executed by the user. Setting up pytpc on Linux distributions involves adding lines to a built-in .bashrc, and the .profile is the macOS equivalent. However, you must first create the .profile file as it is not included on Mac systems by default. **Create and open a plain text file titled `.profile.txt` in your home directory**. This file will not appear in your home directory unless you enter 'ls -a' in the terminal to show hidden files.
+
+.. note::
+
+   Remember to use the correct call to python/python3/python3.6 and pip/pip3/pip3.6 depending how your Python 3.6 was installed. This is neccesary to install software in the correct locations for the remainder of these instructions.
+
+
+2) Install Clang and virtualenvwrapper
+--------------------------------------
+
+a) Install/Update Clang (macOS Only)
+************************************
+
+`Clang <https://clang.llvm.org/>`__ is a compiler front-end that supports multiple languages. Although an older version is already installed on your system, install the newest version using brew:
+
+.. code-block:: shell
+
+   brew install llvm
+
+To make this most recent version of clang the default compiler on your system (and enable OpenMP support) paste the following lines into the .profile file created above:
+
+.. code-block:: shell
+
+   export CC=/usr/local/Cellar/llvm/4.0.0_1/bin/clang-4.0
```

Comment:
Make these paths use the `$(brew --prefix llvm)` command from below. I might write it like this:
```shell
LLVM_PATH=$(brew --prefix llvm)
export CC=${LLVM_PATH}/bin/clang
export CXX=${LLVM_PATH}/bin/clang++
```
That way it doesn't depend on where it installed LLVM (which is `/usr/local/opt/llvm` on my system, actually), and it only runs the slow `brew --prefix llvm` command once.